superphy/prairiedog
# Creates all required files except the node_labels (#28)

* Save graph after benchmark

* Decrease log levels for Kmer parsing

* Benchmark with 0.01% of available genomes first

* 0.001%

* Use underscores instead of : for filenames

* Fix kmer count when creating graph

* Start doing some munging for the format we need

* About to integrate GraphRef as a component to KmerGraph

* Restructure code to suit the DiffPool format, will break a bunch of tests

* Restructure code to suit the DiffPool format, will break a bunch of tests

* Comment out GraphRef.append() to see what tests pass

* Change tests to be based of subgraphs

* Fix output folder creation

* Check if the kmer_map is bugging out

* Use a shutdown()

* Fix node_id iteration

* Fix node_id iteration

* Unique count is slow

* Use a set

* Use a set

* Contextmanage the ProcessPoolExecutor

* Fix using add instead of update

* Add a check against unique_kmer count

* Add a check against unique_kmer count

* Fix linting errors

* Add a check against KmerGraph node ids

* Fix actual NetworkX graph creation

* Use shortened files for KmerGraph tests

* Use shortened files for KmerGraph tests

* Try to fix class method patching

* Try to fix class method patching

* Try to fix class method patching

* Check that the # of unqiue kmers == # nodes

* Check that the # of unqiue kmers == # nodes

* Add the initial framework for creating graph indicators/labels and node labels/attributes

* Add pandas as a req

* Integrate coveralls

* Setup default samples/

* Fix syntax errors

* Fix syntax errors

* Fix import MIC data

* Fix import MIC data

* Was ignoring the mic df.pkl

* Dont try to label a test file

* Fix cli test

* Fix cli test

* Fix flake8 error

* Fix import

* Check whats going on with codecov

* Move codecov call to circleci instead of tox

* Move codecov call to circleci instead of tox

* Fix? pytest-cov

* Fixes codecov

* Cache both requirements.txt and requirements_dev.txt

* Move GRef and SubgraphRef to their own files

* Add a check against the KMERS_graph_indicator.txt file

* Use shortened files for graph output tests

* Check CSVs can be read

* See if setting config.MIC_CSV directly works instead of patching

* Fix KMERS_graph_indicator.txt output

* Fix KMERS_graph_indicator.txt output

* Test KMERS_node_labels.txt and KMERS_node_attributes.txt output

* Fix dictionary write out

* Fix dictionary write out

* Check that we havent assigned some kmer id > 4^k

* Switch to json lib to dump dictionaries

* Switch to json lib to dump dictionaries

* Back to file writing, json lib doesnt accept int as dict keys

* Write out adj matrix

* Write out adj matrix

* Use abstract method
